metrics/prometheus.go

```diff
@@ -0,0 +1,261 @@
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+	}
+	counter := prometheus.NewCounter(opts)
+	registry.MustRegister(counter)
+	return counter
+}
+
+func newHistogram(cfg config.PrometheusMetrics, registry *prometheus.Registry, name, help string, buckets []float64) prometheus.Histogram {
+	opts := prometheus.HistogramOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+		Buckets:   buckets,
+	}
+	histogram := prometheus.NewHistogram(opts)
+	registry.MustRegister(histogram)
+	return histogram
+}
+
+func newGaugeMetric(cfg config.PrometheusMetrics, registry *prometheus.Registry, name string, help string) prometheus.Gauge {
+	opts := prometheus.GaugeOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+	}
+	gauge := prometheus.NewGauge(opts)
+	registry.MustRegister(gauge)
+	return gauge
+}
+
+func newHistogramVector(cfg config.PrometheusMetrics, registry *prometheus.Registry, name, help string, labels []string, buckets []float64) *prometheus.HistogramVec {
+	opts := prometheus.HistogramOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+		Buckets:   buckets,
+	}
+	histogram := prometheus.NewHistogramVec(opts, labels)
+	registry.MustRegister(histogram)
+	return histogram
+}
+
+/**************************************************
+ *	DEPECRATED Functions to record metrics
+ **************************************************/
+func (m PrometheusMetrics) Export(cfg config.Metrics) {
+}
+
+/**************************************************
+ *	NEW Functions to record metrics
+ **************************************************/
+func (m *PrometheusMetrics) RecordPutRequest(status string, duration *time.Time) {
+	incCounterInVector(m.Puts.RequestStatus, "status", status, []string{"add", "error", "bad_request"})
+	incDuration(m.Puts.Duration, duration)
+}
+
+func (m *PrometheusMetrics) RecordGetRequest(status string, duration *time.Time) {
+	incCounterInVector(m.Gets.RequestStatus, "status", status, []string{"add", "error", "bad_request"})
+	incDuration(m.Gets.Duration, duration)
+}
+
+func (m *PrometheusMetrics) RecordPutBackendRequest(status string, duration *time.Time, sizeInBytes float64) {
+	incDuration(m.PutsBackend.Duration, duration)
+	incCounterInVector(m.PutsBackend.PutBackendRequests, "label", status, []string{"add", "json", "xml", "invalid_format", "defines_ttl", "error"})
+	incSize(m.PutsBackend.RequestLength, sizeInBytes)
+}
+
+func (m *PrometheusMetrics) RecordGetBackendRequest(status string, duration *time.Time) {
+	incCounterInVector(m.GetsBackend.RequestStatus, "status", status, []string{"add", "error", "bad_request"})
+	incDuration(m.GetsBackend.Duration, duration)
+}
+
+func (m *PrometheusMetrics) RecordConnectionMetrics(label string) {
+	if label == "add" {
+		m.Connections.ConnectionsOpened.Inc() //change this for  Gauge if you have time
```

Comment:
No need to change. I think a counter makes sense here.

Reply:
Since the number of connections increases and decreases, a Gauge was implemented. I just forgot to remove the comments.